mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java

```diff
@@ -354,8 +357,13 @@ private void processIntent(Intent intent) {
         switch (action) {
             case NfcAdapter.ACTION_NDEF_DISCOVERED:
             case Intent.ACTION_VIEW:
-                mPendingNfcData = intent.getData();
-                openPendingNfcPageIfNeeded();
+                NfcTag tag = NfcTag.fromTagData(intent.getData());
+
+                if (tag.mustOpenSitemap()) {
+                    mPendingOpenSitemapUrl = tag.sitemap();
+                    openPendingSitemapIfNeeded();
+                }
+
```

Comment:
This `case` block - AFAICT - won't be executed anymore since the respective intent filter was removed from the manifest and thus can be removed.

Reply:
I think it will be used. Don't we tell the nfc adapter to get all tags? I've to check that. https://github.com/openhab/openhab-android/blob/768bc546e43dab81a53828ec70b4ce0dbc58ce7d/mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java#L412-L418

Reply:
# [1302](url)

Reply:
Didn't think of that, but can't we route that one towards the NfcReceiveActivity as well?

Reply:
I'd like to keep this here as I don't want to start an invisible activity only to re-launch the current one. By using the function `NfcTag#mustOpenSitemap()` there's nearly no duplicate code.